clients/mobile/ios/TodoList/ContentView.swift

```diff
@@ -8,17 +8,91 @@
 import SwiftUI
 
 struct ContentView: View {
-    var body: some View {
-        VStack {
-            Image(systemName: "globe")
-                .imageScale(.large)
-                .foregroundStyle(.tint)
-            Text(sayHi())
+  @State private var errorMessage: String = ""
+  @State private var newTaskName: String = ""
+  @State private var showError: Bool = false
+  @State private var todolist: TodoListDto?
+
+  var client = Client()
+
+  var body: some View {
+    NavigationView {
+      VStack {
+        List {
+          ForEach(todolist?.tasks ?? [], id: \.index) { task in
+            HStack {
+              Text(task.name)
+
+              Spacer()
+
+              Toggle(
+                "",
+                isOn: Binding(
+                  get: { task.status == .completed },
+                  set: { _, _ in
+                    Task {
+                      do {
+                        try await client.completeTask(index: task.index)
+                        todolist = try await client.getTodoList()
+                      } catch {
+                        handleError(error)
+                      }
+                    }
+                  })
+              )
+              .labelsHidden()
+            }
+          }
+        }
+
+        HStack {
+          TextField("Enter task name", text: $newTaskName)
+            .textFieldStyle(RoundedBorderTextFieldStyle())
+            .padding()
+
+          Button(
+            action: {
+              Task {
+                do {
+                  try await client.addTask(name: newTaskName)
+                  todolist = try await client.getTodoList()
+                  newTaskName = ""
+                } catch {
+                  handleError(error)
+                }
+              }
+            },
+            label: {
+              Text("Add")
+            }
+          )
+          .padding()
         }
```

Comment:
**Suggestion:** Consider extracting the task completion and addition logic into separate functions for better readability and maintainability. [maintainability]
```suggestion
              Toggle(
                  "",
                  isOn: Binding(
                      get: { task.status == .completed },
                      set: { _, _ in
                          completeTask(task: task)
                      })
                  )
                  .labelsHidden()
              Button(
                  action: {
                      addTask()
                  },
                  label: {
                      Text("Add")
                  }
              )
              // ...
              private func completeTask(task: Task) {
                  Task {
                      do {
                          try await client.completeTask(index: task.index)
                          todolist = try await client.getTodoList()
                      } catch {
                          handleError(error)
                      }
                  }
              }
              private func addTask() {
                  Task {
                      do {
                          try await client.addTask(name: newTaskName)
                          todolist = try await client.getTodoList()
                          newTaskName = ""
                      } catch {
                          handleError(error)
                      }
                  }
              }
```